modules/rgbd/src/normal.cpp

```diff
@@ -836,6 +837,7 @@ multiply_by_K_inv(const Matx<T, 3, 3> & K_inv, U a, U b, U c, Vec<T, 3> &res)
 
     // Get the normals
     normals_out.create(points3d_ori.size(), CV_MAKETYPE(depth_, 3));
+    normals_out.setTo(0);
```

Comment:
In the linemod case, don't seems that every normal is modified [in this `for`](https://github.com/Itseez/opencv_contrib/blob/master/modules/rgbd/src/normal.cpp#L437) but there is a border `r` ignored.
On the contrary in the test all the normals are checked [here](https://github.com/Itseez/opencv_contrib/blob/master/modules/rgbd/test/test_normal.cpp#L327) so the `setTo` influences it.
Since normals should have NaN values on undefined values, I think I should change that `setTo` in this way:
```
  if (depth_ == CV_32F)
      normals_out.setTo(std::numeric_limits<float>::quiet_NaN());
    else
      normals_out.setTo(std::numeric_limits<double>::quiet_NaN());
```

do you agree?